### PR TITLE
Faster loading

### PR DESCRIPTION
**Pull request by bpinto**
- Initial work for supporting plugin dependencies.
- Source `init.fish` over `$package_name.fish`.

### CODE REVIEW

File: init.fish [hunk 1]
```diff
@@ -41,16 +41,20 @@ source $OMF_CONFIG/before.init.fish ^/dev/null
...
 
-  autoload $path $path/completions
-  source $path/(basename $path).fish ^/dev/null
-    and emit init_(basename $path) $path
+  contains -- $name $OMF_IGNORE; and continue
+  require $name
 end
 
+# Autoload theme
+autoload {$OMF_PATH,$OMF_CONFIG}/themes/(cat $OMF_CONFIG/theme)
```
Comment (bpinto) [inline init.fish]: feature

The idea is to support overwriting specific functions on your theme/plugin. So if you correctly split functions on individual files, one have the opportunity to change a specific part of it.

Comment (bobthecow) [inline init.fish]:
```
function autoload_one -d "autoload the first directory which exists in the list of directories"
  for path in $argv
    autoload $path; and return
  end
end
```

then we could

```
autoload_one {$OMF_CONFIG,$OMF_PATH}/themes/(cat $OMF_CONFIG/theme)
```

?

(note that the OMF_PATH/OMF_CONFIG are swapped, so the local one wins if two have the same name)

Comment (bobthecow) [inline init.fish]: yeah. kind of makes sense. also makes things potentially crazy :)

Comment (bpinto) [inline init.fish]: That's true, IMHO the benefits of allowing this are bigger. We have supported this for a long time (not sure in years) and I don't remember we having a problem with it.

File: pkg/omf/init.fish [hunk 2]
```diff
@@ -0,0 +1,3 @@
+function init -a path --on-event init_omf
```
Comment (bobthecow) [inline pkg/omf/init.fish]: i'm not a fan of how our plugins all leak `init` functions :(

**Comment by bobthecow** [comment]
:+1:

File: pkg/omf/init.fish (hunk 2 above)
Comment (bpinto) [inline pkg/omf/init.fish]: Yeah me neither... this kinda defeats the purpose of having the `init_omf` event. BUT, I couldn't think of a better and easier to use idea yet...

File: init.fish (hunk 1 above)
Comment (bobthecow) [inline init.fish]: :+1:

**Pull request merged**